### PR TITLE
Refactor examples.js to use async/await

### LINKED ISSUE #1693: Refactor `fetch-examples` script to only use async/await
#### Nature of issue?

- Existing feature enhancement

#### Feature enhancement details:

The file [`examples.js`](https://github.com/processing/p5.js-web-editor/blob/develop/server/scripts/examples.js) has been edited by many people over the years and could use a refactor. It uses callbacks, Promises, and async/await, and I think that it should be refactored to just use async/await.

### PR DESCRIPTION
**Pull request by Swapnil-2001**
Fixes #1693 

I have verified that this pull request:

* [x] has no linting errors (`npm run lint`)
* [x] is from a uniquely-named feature branch and has been rebased on top of the latest `develop` branch. (If I was asked to make more changes, I have made sure to rebase onto `develop` then too)
* [x] is descriptively named and links to an issue number, i.e. `Fixes #123`

@catarak please let me know if the changes are okay.

### CODE REVIEW

**Comment by catarak** [comment]
Thanks for working on this! Your changes look great. One thing I would also suggest changing is the reliance on the library `async` (and the function `eachSeries`) which is used to make using callback functions easier. I believe you could get rid of it and just use the library `q`!

**Comment by Swapnil-2001** [comment]
Okay, I'll be changing that!

**Comment by Swapnil-2001** [comment]
Could you please go through the changes? This is my first time working with the library `q` so I'm not 100% sure!

File: server/scripts/examples.js [hunk 1]
```diff
@@ -236,125 +234,123 @@ function createProjectsInP5user(projectsInAllCategories) {
...
-            projectCallback();
-          });
-        }, (categoryErr) => {
-          categoryCallback();
+        newProject.save((saveErr, savedProject) => {
+          if (saveErr) throw saveErr;
+          console.log(`Created a new project in p5 user: ${savedProject.name}`);
         });
-      }, (examplesErr) => {
-        process.exit();
```
Comment (catarak) [inline server/scripts/examples.js:347]: You removed this `process.exit`, so now when you run the `fetch-examples` script, it hangs when it's done and never closes. Could you add this back?

**Review by catarak — CHANGES_REQUESTED** [review]

File: server/scripts/examples.js (hunk 1 above)
Comment (Swapnil-2001) [reply server/scripts/examples.js:347]: I've added it back. Please let me know if there is anything else that needs to be changed.

**Comment by catarak** [comment]
Looking at this again, I also noticed that the `mongoose` queries (e.g. `User.findOne`) use callbacks—could these use `async/await` as well?

**Comment by Swapnil-2001** [comment]
Of course!

Also, I replaced `remove` with `deleteOne`, since `remove` is deprecated.

**Pull request merged**